src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -539,77 +539,15 @@ internal bool RequiresRefSafetyRulesAttribute()
                 _lazyRequiresRefSafetyRulesAttribute = value.ToThreeState();
             }
             return _lazyRequiresRefSafetyRulesAttribute.Value();
 
-            static bool namespaceIncludesRefs(NamespaceSymbol ns)
+            static bool isFeatureDisabled(CSharpCompilation compilation)
             {
-                foreach (var member in ns.GetMembersUnordered())
-                {
-                    switch (member.Kind)
-                    {
-                        case SymbolKind.Namespace:
-                            if (namespaceIncludesRefs((NamespaceSymbol)member))
-                            {
-                                return true;
-                            }
-                            break;
-                        case SymbolKind.NamedType:
-                            if (typeIncludesRefs((NamedTypeSymbol)member))
-                            {
-                                return true;
-                            }
-                            break;
-                        default:
-                            throw ExceptionUtilities.UnexpectedValue(member.Kind);
-                    }
-                }
-                return false;
-            }
-
-            static bool typeIncludesRefs(NamedTypeSymbol type)
-            {
-                if (type.IsRefLikeType)
-                {
-                    return true;
-                }
-                foreach (var member in type.GetMembersUnordered())
-                {
-                    switch (member.Kind)
-                    {
-                        case SymbolKind.NamedType:
-                            if (typeIncludesRefs((NamedTypeSymbol)member))
-                            {
-                                return true;
-                            }
-                            break;
-                        case SymbolKind.Method:
-                            if (methodIncludesRefs((MethodSymbol)member))
-                            {
-                                return true;
-                            }
-                            break;
-                    }
-                }
-                return false;
-            }
-
-            static bool methodIncludesRefs(MethodSymbol method)
-            {
-                if (method.RefKind != RefKind.None ||
-                    method.ReturnType.IsRefLikeType)
-                {
-                    return true;
-                }
-                return method.Parameters.Any(p => parameterIncludesRef(p));
-            }
-
-            static bool parameterIncludesRef(ParameterSymbol parameter)
-            {
-                return parameter.RefKind != RefKind.None ||
-                    parameter.Type.IsRefLikeType;
+                var options = (CSharpParseOptions?)compilation.SyntaxTrees.FirstOrDefault()?.Options;
+                return options?.Features?.ContainsKey("noRefSafetyRulesAttribute") == true;
```

Comment:
nit: do we need any kind of documentation for this feature flag, or will it just be explained on a case-by-case basis?

Reply:
As we discussed offline, `-features:noRefSafetyRulesAttribute` will disable emitting `[module: RefSafetyRules]` unconditionally, and is only intended for use in tests. It is not intended for use outside of tests.